tinyDA/chain.py

```diff
@@ -715,6 +717,8 @@ def sample(self, iterations, progressbar=True):
                 # state-dependent error model has not been implemented.
                 elif self.adaptive_error_model == "state-dependent":
                     pass
+                else:
+                    raise NameError("Adaptive error model can only be state-dependent, state-independent or None.")
```

Comment:
I don't think we need to check this at every iteration, since this already lives in the `__init__` method.